Src/FluentAssertions/Common/TypeMemberReflector.cs

```diff
@@ -123,10 +140,10 @@ private static List<FieldInfo> GetFieldsFromHierarchy(Type typeToReflect, Member
             members.InsertRange(0, newPropertyInfos);
         }
 
-        return members;
+        return members.ToArray();
```

Comment:
Because Qodana wanted me to use a more abstract type than `List<TMemberINfo` as the return value of `GetInterfaceMembers`. I could try to change it to `IEnumerablt<T>`